packages/hint/src/lib/hint-context.ts

```diff
@@ -138,7 +140,8 @@ export class HintContext<E extends Events = Events> {
             codeSnippet || sourceCode || '',
             position,
             message,
-            resource
+            resource,
+            options.codeLanguage
```

Comment:
Not sure we need to do it in this PR, but it'd be nice to refactor `engine.report` to take something similar to `ReportOptions` instead of continuing to add extra parameters...

Reply:
I've opened https://github.com/webhintio/hint/issues/2276

Reply:
I think this is a breaking change, so it would be nice to have it sooner rather than later.

Reply:
I agree, we have quite a few breaking changes already. The blogpost is going to be quite big...

Reply:
"How we rebuilt webhint from the ground up..." 🤣